commands/run.sh

```diff
@@ -0,0 +1,542 @@
+  # be overridden to host per limitations of docker
+  # https://docs.docker.com/engine/security/userns-remap/#user-namespace-known-limitations
+  if [[ "${BUILDKITE_PLUGIN_DOCKER_PRIVILEGED:-false}" =~ ^(true|on|1)$ ]] ; then
+      args+=("--userns" "host")
+  else
+      args+=("--userns" "${BUILDKITE_PLUGIN_DOCKER_USERNS:-}")
+  fi
+fi
+
+# Mount ssh-agent socket and known_hosts
+if [[ ! "${BUILDKITE_PLUGIN_DOCKER_MOUNT_SSH_AGENT:-false}" = 'false' ]] ; then
+  if [[ -z "${SSH_AUTH_SOCK:-}" ]] ; then
+    echo "+++ 🚨 \$SSH_AUTH_SOCK isn't set, has ssh-agent started?"
+    exit 1
+  fi
+  if [[ ! -S "${SSH_AUTH_SOCK}" ]] ; then
+    echo "+++ 🚨 There file at ${SSH_AUTH_SOCK} does not exist or is not a socket, has ssh-agent started?"
+    exit 1
+  fi
+
+  if [[ "${BUILDKITE_PLUGIN_DOCKER_MOUNT_SSH_AGENT:-''}" =~ ^(true|on|1)$ ]]; then
+    MOUNT_PATH=/root
+  else
+    MOUNT_PATH="${BUILDKITE_PLUGIN_DOCKER_MOUNT_SSH_AGENT}"
+  fi
+
+  args+=(
+    "--env" "SSH_AUTH_SOCK=/ssh-agent"
+    "--volume" "${SSH_AUTH_SOCK}:/ssh-agent"
+    "--volume" "${HOME}/.ssh/known_hosts:${MOUNT_PATH}/.ssh/known_hosts"
+  )
+fi
+
+# Handle the mount-buildkite-agent option
+if [[ "${BUILDKITE_PLUGIN_DOCKER_MOUNT_BUILDKITE_AGENT:-$mount_agent_default}" =~ ^(true|on|1)$ ]] ; then
+  if [[ -z "${BUILDKITE_AGENT_BINARY_PATH:-}" ]] ; then
+    if is_windows; then
+      echo -n "+++ 🚨 You must specify the variable BUILDKITE_AGENT_BINARY_PATH to mount the agent in Windows"
+      exit 1
+    elif ! command -v buildkite-agent >/dev/null 2>&1 ; then
+      echo -n "+++ 🚨 Failed to find buildkite-agent in PATH to mount into container, "
+      echo "you can disable this behaviour with 'mount-buildkite-agent:false'"
+    else
+      BUILDKITE_AGENT_BINARY_PATH=$(command -v buildkite-agent)
+    fi
+  fi
+fi
+
+# Mount buildkite-agent if we have a path for it
+if [[ -n "${BUILDKITE_AGENT_BINARY_PATH:-}" ]] ; then
+  args+=(
+    "--env" "BUILDKITE_JOB_ID"
+    "--env" "BUILDKITE_BUILD_ID"
+    "--env" "BUILDKITE_AGENT_ACCESS_TOKEN"
+    "--volume" "$BUILDKITE_AGENT_BINARY_PATH:${agent_mount_folder}"
+  )
+fi
+
+# Parse extra env vars and add them to the docker args
+while IFS='=' read -r name _ ; do
+  if [[ $name =~ ^(BUILDKITE_PLUGIN_DOCKER_ENVIRONMENT_[0-9]+) ]] ; then
+    args+=( "--env" "${!name}" )
+  fi
+done < <(env | sort)
+
+# Parse host mappings and add them to the docker args
+while IFS='=' read -r name _ ; do
+  if [[ $name =~ ^(BUILDKITE_PLUGIN_DOCKER_ADD_HOST_[0-9]+) ]] ; then
+    args+=( "--add-host" "${!name}" )
+  fi
+done < <(env | sort)
+
+# Privileged container
+if [[ "${BUILDKITE_PLUGIN_DOCKER_PRIVILEGED:-false}" =~ ^(true|on|1)$ ]] ; then
+    args+=( "--privileged" )
+fi
+
+if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_ENV_FILE; then
+  # don't warn about result being unassigned (plugin_read_list_into_result assigns to global result variable)
+  # shellcheck disable=SC2154
```

Comment:
All these disabling of checks can be easily avoided if you just source the shared library that defines `plugin_read_list_into_result` at the top:

```
DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"

# shellcheck source=lib/shared.bash
. "$DIR/../lib/shared.bash"
```